sql/catalyst/src/test/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatchSuite.java

```diff
@@ -127,30 +127,43 @@ public void emptyBatch() throws Exception {
     try (RowBasedKeyValueBatch batch = RowBasedKeyValueBatch.allocate(keySchema,
         valueSchema, taskMemoryManager, DEFAULT_CAPACITY)) {
       Assert.assertEquals(0, batch.numRows());
+
+      boolean asserted = false;
       try {
         batch.getKeyRow(-1);
-        Assert.fail("Should not be able to get row -1");
```

Comment:
Same problem -- if it 'failed' it would still pass as the fail() is caught by the catch block